app/api/src/docker/common/utils.js

```diff
@@ -35,7 +48,101 @@ async function runCommand(command, timeout = 5000) {
   })
 }
 
+function makeHash(script) {
+  return createHash('sha256').update(script).digest('hex')
+}
+
+async function checkIfAlreadyExists(params, s3) {
+  try {
+    const objectHead = await s3.headObject(params).promise()
+    const consoleMessage = getConsoleMessageFromMetadata(objectHead.Metadata)
+    console.log('consoleMessage', consoleMessage)
+    return { isAlreadyInBucket: true, consoleMessage }
+  } catch (e) {
+    console.log("couldn't find it", e)
+    return { isAlreadyInBucket: false }
+  }
+}
+
+function getObjectUrl(params, s3) {
+  const HALF_HOUR = 1800
+  return s3.getSignedUrl('getObject', {
+    ...params,
+    Expires: HALF_HOUR,
+  })
+}
+
+function loggerWrap(handler) {
+  return (req, _context, callback) => {
+    try {
+      return handler(req, _context, callback)
+    } catch (e) {
+      console.log('error in handler', e)
+    }
+  }
+}
+
+async function storeAssetAndReturnUrl({
```

Comment:
After a part is successful generated the following steps is common to both OpenSCAD and CadQuery so threw it in a function, should probably be cleaned up a little.